internal/ethapi/api.go

```diff
@@ -1006,6 +1006,9 @@ func (s *PublicTransactionPoolAPI) GetTransactionReceipt(hash common.Hash) (map[
 		return nil, nil
 	}
 	receipt, _, _, _ := core.GetReceipt(s.b.ChainDb(), hash) // Old receipts don't have the lookup data available
+	if receipt == nil {
+		return nil, nil
+	}
```

Comment:
Looks good to me in general, but I would suggest that you signal an error here, something like: 
```
     return nil, fmt.Errorf("Missing receipt for hash %v", hash)
```

Reply:
done.
for nil transaction response too.